cpp/src/join/join_kernels.cuh

```diff
@@ -130,22 +129,14 @@ __global__ void compute_join_output_size( multimap_type const * const multi_map,
                                           gdf_size_type* output_size)
 {
 
-  __shared__ gdf_size_type block_counter;
-  block_counter=0;
-  __syncthreads();
-
-#if defined(CUDA_VERSION) && CUDA_VERSION >= 9000
-  __syncwarp();
-#endif
+  gdf_size_type thread_counter {0};
+  const gdf_size_type start_idx {(gdf_size_type) (threadIdx.x + blockIdx.x * blockDim.x)};
```

Comment:
cast is necessary because we are converting from an `unsigned int` to `int`. How about
```
const gdf_size_type start_idx {static_cast<gdf_size_type>(threadIdx.x + blockIdx.x * blockDim.x)}
```

Reply:
I'd just use normal assignment instead of brace initialization. Then the conversion will happen implicitly.
```suggestion
  const gdf_size_type start_idx = threadIdx.x + blockIdx.x * blockDim.x;
```

Reply:
Resolved in https://github.com/rapidsai/cudf/pull/2293/commits/bbcd13d807c3ddb12aa6a1d0dad22030489181b6.